### PR TITLE
Fixes #14980: terraform configuration v1

### PR DESCRIPTION
**Pull request by el-k0022**
https://issues.rudder.io/issues/14980

### CODE REVIEW

**Comment by el-k0022** [comment]
Commit modified

**Comment by el-k0022** [comment]
Commit modified

**Comment by el-k0022** [comment]
Commit modified

File: scripts/terraform/README.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* This example creates AWS VPC resources, making a VPC in eu-west region ans crefate also subnet in 
+VPC. You have possibility to create a set of subnets in differents availability zones.
+
+NOTE: Region : container module for all of the network resources within a region. This is instantiated once per region.
+
+NOTE: Subnet : presents a subnet within a given availability zone. This is instantiated twice per region, using the first availability zone supported within the target AWS account.
+
+== Installation 
+
+* Terraform must firdt be installed in your machine, it's distributed as a binary package. To install
```
Comment (peckpeck) [inline scripts/terraform/README.adoc]: first

File: scripts/terraform/README.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+or 
+```
+terraform destroy
+```
+* The terraform plan command is used to create an execution plan. Terraform performs a refresh, unless explicitly disabled, and then determines what actions are necessary to achieve the desired state specified in the configuration files.
+
+* The terraform apply command is used to apply the changes required to reach the desired state of the configuration, or the pre-determined set of actions generated by a terraform plan execution plan.
+
+* The terraform destroy command is used to destroy the Terraform-managed infrastructure.
+
```
Comment (peckpeck) [inline scripts/terraform/README.adoc]: This is generic terraform doc.
We want something specific to this configuration here, so something like:
- go to scripts/terraform
- run terraform plan
- run terraform deploy

Comment (peckpeck) [reply scripts/terraform/README.adoc]: + edit the conf to change the ssh key

File: scripts/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  default = "10.0.2.0/24"
+}
+
+# Configurations for a given provider name
+##########
+# aws auth
+
+provider "aws" {
+    region     = "${var.aws_region}"
+    access_key = ""
```
Comment (peckpeck) [inline scripts/terraform/main.tf]: add a comment here to say that this must be changed

File: scripts/terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    values    = ["hvm"]
+  }
+  owners      = ["099720109477"]
+}
+
+# ssh keys 
+
+resource "aws_key_pair" "admin" {
+  key_name   = "admin.key"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDO0cD6HK2cPutURkX11lR1DG1+/Ry1t3Jp+RCTybp3wFVHCKitCcwqyZEhGuviTLJgCaL71xiKjkKjt9VoIEi7FnJ+gQdRC/OTc6RwhUYdox5mgOeM/1erb1949eD0OE2mVAw/al+xPZBTOejTQzRgOvIfEP7y0WlM3dAOqAgTCvJnysxxmHUh7skCt6mxLK5tpTmnpEb9bFXTIcbMKtKhLGkOImOUeVTRQePS4E2IImLouAl+QFbmbfxPM6A962FSGL1iZXF/Pkkm8IfSaUybkx+mcV4zD/l8BfYhNYJSujVJn/OoE99/2escJFoJBvgkRVNkAf583wrlx1M9k3Il nel@nel"
```
Comment (peckpeck) [inline scripts/terraform/main.tf]: add a comment here to say that this must be changed

File: scripts/terraform/main.tf [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    security_groups = [
+        "${aws_security_group.allow_remote_admin.id}"
+    ]
+    
+    provisioner "remote-exec" {
+    	inline = [ 
+	"sudo wget https://repository.rudder.io/tools/rudder-setup",
+	"sudo chmod +x rudder-setup",
+	"sudo ./rudder-setup setup-server latest"
```
Comment (peckpeck) [inline scripts/terraform/main.tf]: Add a comment to say that it is possible to replace "latest" with a specific rudder version

File: scripts/terraform/terraform.tfstate [hunk 6]
```diff
@@ -0,0 +1,357 @@
+{
```
Comment (peckpeck) [inline scripts/terraform/terraform.tfstate]: I this this file is specifically created by terraform for your tests and should not be commited

**Comment by el-k0022** [comment]
Commit modified

File: README.md [hunk 7]
```diff
@@ -1,6 +0,0 @@
-rudder-tools
```
Comment (amousset) [inline README.md]: You should not remov emain README.

**Comment by el-k0022** [comment]
Commit modified

**Comment by el-k0022** [comment]
Commit modified

**Comment by el-k0022** [comment]
Commit modified

**Pull request merged**